spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinSpanListener.java

```diff
@@ -189,17 +182,7 @@ private void addZipkinBinaryAnnotations(io.zipkin.Span.Builder zipkinSpan,
 		}
 	}
 
-	private static long hash(String string) {
-		long h = 1125899906842597L;
-		if (string == null) {
-			return h;
-		}
-		int len = string.length();
-
-		for (int i = 0; i < len; i++) {
-			h = 31 * h + string.charAt(i);
-		}
-		return h;
+	private long valueOrDefault(Long value) {
```

Comment:
added this issue about the general problem https://github.com/spring-cloud/spring-cloud-sleuth/issues/114